platform/posix/transport/CMakeLists.txt

```diff
@@ -1,51 +1,66 @@
 # Include filepaths for source and include.
 include(${PLATFORM_DIR}/posix/posixFilePaths.cmake)
 
+# The transport_interface.h in coreMQTT is the single source of truth,
+# so make sure the submodule is cloned.
+if( NOT EXISTS ${MODULES_DIR}/standard/coreMQTT/source )
+    # Attempt to clone coreMQTT.
+    if( ${BUILD_CLONE_SUBMODULES} )
+        clone_mqtt()
+    else()
+        message( FATAL_ERROR "The required submodule coreMQTT does not exist. Either clone it manually, or set BUILD_CLONE_SUBMODULES to 1 to automatically clone it during build." )
+    endif()
+endif()
+
+set( TRANSPORT_INTERFACE_INCLUDE_DIR
+     ${MODULES_DIR}/standard/coreMQTT/source/portable )
+
 # Create target for sockets utility.
 add_library( sockets_posix
                 ${SOCKETS_SOURCES} )
 
-target_include_directories( sockets_posix 
-                              PUBLIC 
+target_include_directories( sockets_posix
+                            PUBLIC
                                 ${COMMON_TRANSPORT_INCLUDE_PUBLIC_DIRS}
-                                ${LOGGING_INCLUDE_DIRS} )
+                                ${LOGGING_INCLUDE_DIRS}
+                                ${TRANSPORT_INTERFACE_INCLUDE_DIR} )
 
 # Create target for plaintext transport.
 add_library( plaintext_posix
-                ${PLAINTEXT_TRANSPORT_SOURCES} )
+             ${PLAINTEXT_TRANSPORT_SOURCES} )
 
-target_link_libraries( plaintext_posix 
-                         PUBLIC 
+target_link_libraries( plaintext_posix
+                       PUBLIC 
                            sockets_posix )
 
 # Create target for POSIX implementation of OpenSSL.
 add_library( openssl_posix
                 ${OPENSSL_TRANSPORT_SOURCES} )
 
 set(OPENSSL_USE_STATIC_LIBS TRUE)
```

Comment:
spacing of this one but it is not a big deal for me tbh

Reply:
Oh missed some thanks. yea the spacing in the file was bothering me lol